src/resolvelib/providers.py

```diff
@@ -13,7 +13,7 @@ def identify(self, dependency):
         raise NotImplementedError
 
     def get_preference(self, resolution, candidates, information):
-        """Produce a sort key for given specification based on preference.
+        """Produce a sort key for given candidate based on preference.
```

Comment:
This should be *requirement*, not candidate. (same below)

Reply:
Ahahahah